nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1277,79 +1285,348 @@ public void run() {
         }
     }
 
-    protected boolean registerRemoteInstance() throws IOException, ConfigurationException, PackageException {
-        String username;
-        String password;
-        String project;
-        String description = "";
+    /**
+     * Prompts for a valid email address according to RFC 822 standards. The remote service may apply stricter
+     * constraints on email validation such as some black listed domains.
+     *
+     * @return The user input. Never null.
+     * @throws ConfigurationException If the user input is read from stdin and is {@code null} or does not match the
+     *             {@code regex}
+     * @throws IOException
+     * @since 8.3
+     */
+    public String promptEmail() throws IOException, ConfigurationException {
```

Comment:
Code looks very similar to `prompt()`, probably can be factored by passing the regex match and the validator as lambdas.